apps/api/src/applicant/entity/ienjob.entity.ts

```diff
@@ -66,4 +68,10 @@ export class IENApplicantJob {
       this.status_audit = sortStatus(this.status_audit);
     }
   }
+  @BeforeInsert()
+  beforeInsert() {
```

Comment:
this job_id is for reference to a real job posting. so I don't think it makes sense to generate a random id.

Reply:
I think having this field blank and not generating one will create more issues than it solves. 
If we don't have it, we'll have trouble differentiating between job competitions. Also, we'll have an awkward edge case when checking for the uniqueness of the ID field. 

Plus we'll need to sync jobs with ATS eventually and having that uuid will allow that process to go smoothly. 

The main case I can against it is in the case of de-duplication or matching with an existing job within ATS. I don't think that's within the scope of our work.

Reply:
I get your point. Another question. randomUUID is a little long and verbose to display. Do we want to abbreviate it, like the first 4 bytes?

Reply:
That works 💯